src/components/course/data/hooks.js

```diff
@@ -157,7 +156,9 @@ export function useCoursePacingType(courseRun) {
   return [pacingType, pacingTypeContent];
 }
 
-export function useCoursePriceForUserSubsidy(activeCourseRun, userSubsidy) {
+export function useCoursePriceForUserSubsidy({
```

Comment:
The original design of `useCoursePriceForUserSubsidy` was to accept a single "generic" `userSubsidy` object that represents which subsidy, if any, would be applied to the course to calculate the price accordingly. 

I don't think this hook should be responsible for determining which subsidy should apply (a license or an offer) to the price, but rather accepts a generic `userSubsidy` (really, `userSubsidyApplicableToCourse` is a more descriptive name) object that represents either a license, an offer (code), or `null` that can be applied to the course.

The offers data we get looks like this:

```json
  "benefit_value": 100
  "catalog": "09a6dd6a-a6f0-4232-aaca-3bd1d81a4197"
  "code": "3NYUCFDVBJGZYL2U"
  "coupon_end_date": "2025-09-24T00:00:00Z"
  "coupon_start_date": "2020-09-24T00:00:00Z"
  "redemptions_remaining": 1
  "usage_type": "Percentage"
}
```

Based on this, we designed the `license-subsidy` endpoint to return a generic interface in its response that could be adopted for offers/codes logic:

```json
{
  "discount_type": "percentage",
  "discount_value": 100,
  "expiration_date": "2021-07-02",
  "start_date": "2020-07-02",
  "status": "activated",
  "subsidy_id": "15bfca2a-ccfe-4c4a-a896-a8709481e119",
}
```

The idea being that as long as `userSubsidy` follows a generic "interface",  we wouldn't need to differentiate between subsidy types in `useCoursePriceForUserSubsidy` or elsewhere, where you could pass in any `userSubsidy` (whether it represents a license, offer, etc.) and the hook would calculate the price accordingly because it expects that `userSubsidy` contains all the data it needs without needing to understand subsidy types, e.g. `discount_type`, `discount_value`.